hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/TokenUpdatePrecompileSuite.java

```diff
@@ -552,13 +552,468 @@ final HapiSpec tokenUpdateSingleFieldCases() {
+                                .hasPauseKey(MULTI_KEY),
+                        contractCall(
+                                        tokenInfoUpdateContract,
+                                        updateToEdKey,
+                                        HapiParserUtil.asHeadlongAddress(asAddress(token.get())),
+                                        spec.registry()
+                                                .getKey(ED25519KEY)
+                                                .getEd25519()
+                                                .toByteArray(),
+                                        2)
+                                .via("updateOnlyFreezeKeyWithEd")
+                                .logged()
+                                .signedBy(contractIdKey, MULTI_KEY)
+                                .payingWith(ACCOUNT)
+                                .gas(GAS_TO_OFFER)
+                                .hasKnownStatus(SUCCESS),
+                        getTokenInfo(TOKEN)
+                                .logged()
+                                .hasTokenType(TokenType.FUNGIBLE_COMMON)
+                                .hasSymbol(sym1)
+                                .hasName(newName)
+                                .hasEntityMemo(newMemo)
+                                .hasTreasury(newTokenTreasury)
+                                .hasAutoRenewAccount(AUTO_RENEW_ACCOUNT)
+                                .hasAutoRenewPeriod(AUTO_RENEW_PERIOD - 1000L)
+                                .hasSupplyType(TokenSupplyType.INFINITE)
+                                .searchKeysGlobally()
+                                .hasAdminKey(contractIdKey)
+                                .hasKycKey(contractIdKey)
+                                .hasFreezeKey(ED25519KEY)
+                                .hasWipeKey(MULTI_KEY)
+                                .hasFeeScheduleKey(MULTI_KEY)
+                                .hasSupplyKey(MULTI_KEY)
+                                .hasPauseKey(MULTI_KEY),
+                        contractCall(
+                                        tokenInfoUpdateContract,
+                                        updateToContractId,
+                                        HapiParserUtil.asHeadlongAddress(asAddress(token.get())),
+                                        HapiParserUtil.asHeadlongAddress(asAddress(spec.registry()
+                                                .getKey(contractIdKey)
+                                                .getContractID())),
+                                        2)
+                                .via("updateOnlyFreezeKey")
+                                .logged()
+                                .signedBy(ED25519KEY, MULTI_KEY)
+                                .payingWith(ACCOUNT)
+                                .gas(GAS_TO_OFFER)
+                                .hasKnownStatus(SUCCESS),
+                        getTokenInfo(TOKEN)
```

Comment:
Since the test is quite large, we could consider adding some comments as dividers for different test scenarios, so we can read the test more easily.

Reply:
Noted. Comments added